tools/Update-ApiVersions.ps1

```diff
@@ -3,124 +3,80 @@
 # INSTRUCTIONS:
```

Comment:
Had to fix this script's handling of child resources when updating apiVersions.  Also did a bit of clean-up, and changing it to be a cmdlet forced me to move functions around to have them defined before being used.  So most of this is just moved code.